### DISCUSSION TITLE
Receiving empty /tf message

### DISCUSSION THREAD

**Discussion by marijapili** [foxglove/community#74] [Q&A] [answered]
Hi,

I am running a desktop version of Foxglove Studio (the source is a websocket on the same network) and seem to have an issue with receiving /tf and /tf_static messages (all other messages work normally). The messages are being published, I used _rostopic echo /tf_ on the source computer and it returned messages normally.

I tested if the problem stems from the websocket itself by connecting to it from terminal using wscat (I even set the compression to 'cbor-raw' to mimic Foxglove) and from there I was able to see some data. In Foxglove, it does recognize the topic type (tf2_msgs/TFMessage), and it shows time of the incoming messages (as if it was receiving something), but the messages themselves are empty.
There are other nodes subscribed to the topic but I tried relaying the message using topic_tools so that the websocket is the only node subscribed but the messages still came in empty.

Another thing worth noting is that there are two websockets running on the source computer (on ports 9090 and 9091). I wasn't able to connect and receive data from the first one because of different compression types, adding the second websocket solved everything except for the problem explained here.

Do you have any suggestions of steps to take to try and solve this problem?

**Comment by jtbandes** [comment]
If you record the messages into a bag file, and load the bag file in studio, do you see the same problem — or can you provide any other steps for us to try and reproduce the problem?  Are there any error messages in the developer console?

**Reply by marijapili** [reply]
If I record it and then load as a bag file, I can see the /tf messages normally. There aren't any errors in the developer console.

I would love to provide more info on how to reproduce the problem, but I haven't been able to reproduce it myself, I am only seeing it for the gazebo simulation I was originally running.

**Comment by marijapili** [comment, marked as answer]
SOLVED - the issue was missing info on the tf2_msgs, I had a package missing (geometry2 to be exact), so the messages couldn't be displayed correctly as there was no message definition
